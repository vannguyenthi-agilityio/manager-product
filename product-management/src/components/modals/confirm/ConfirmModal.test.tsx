// Libraries
import { fireEvent, render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

// Components
import ConfirmModal from '.';

const onClose = jest.fn();
const onClick = jest.fn();

const baseProps = {
  isOpen: true,
  onClose,
  onClick
};

const setupDefault = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps
  };

  return render(<ConfirmModal {...props} />);
};

describe('ConfirmModal component with Snapshot', () => {
  test('should render correctly', () => {
    setupDefault();

    const formDetail = screen.getByTestId('confirm-modal');

    expect(formDetail).toMatchSnapshot();
  });
});

describe('ConfirmModal component', () => {
  test('should call onClick when click delete button', async () => {
    jest.useFakeTimers();
    setupDefault();
    const confirmBtn = screen.getByRole('button', { name: /Close/i });

    await act(async () => {
      fireEvent.click(confirmBtn);
      jest.runOnlyPendingTimers();
    });

    expect(onClick).toBeCalledTimes(1);
  });
});

const setupDelete = (moreProps = {}) => {
  const props = {
    type: 'delete',
    title: 'Delete Product',
    content: 'Are you sure you want delete this product? This action cannot be undone',
    ...baseProps,
    ...moreProps
  };

  return render(<ConfirmModal {...props} />);
};

describe('ConfirmModal component', () => {
  test('should call onClick when click delete button', async () => {
    jest.useFakeTimers();
    setupDelete();
    const confirmBtn = screen.getByRole('button', { name: /Delete/i });

    await act(async () => {
      fireEvent.click(confirmBtn);
      jest.runOnlyPendingTimers();
    });

    expect(onClick).toBeCalledTimes(2);
  });
});
