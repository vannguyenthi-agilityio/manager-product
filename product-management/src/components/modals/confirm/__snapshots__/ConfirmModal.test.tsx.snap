// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal component with Snapshot should render correctly 1`] = `
<section
  aria-describedby="chakra-modal--body-:r0:"
  aria-labelledby="chakra-modal--header-:r0:"
  aria-modal="true"
  class="chakra-modal__content css-6yiygt"
  data-testid="confirm-modal"
  id="chakra-modal-:r0:"
  role="dialog"
  style="opacity: 0; transform: scale(0.95) translateZ(0);"
  tabindex="-1"
>
  <header
    class="chakra-modal__header css-1ehdivf"
    id="chakra-modal--header-:r0:"
  >
    <div
      class="chakra-stack css-8g8ihq"
    >
      <svg
        class="chakra-icon css-2yv1oa"
        focusable="false"
        viewBox="0 0 56 56"
      >
        <rect
          fill="#FEE4E2"
          height="48px"
          rx="24"
          width="48px"
          x="4"
          y="4"
        />
        <path
          d="M32 22V21.2C32 20.0799 32 19.5198 31.782 19.092C31.5903 18.7157 31.2843 18.4097 30.908 18.218C30.4802 18 29.9201 18 28.8 18H27.2C26.0799 18 25.5198 18 25.092 18.218C24.7157 18.4097 24.4097 18.7157 24.218 19.092C24 19.5198 24 20.0799 24 21.2V22M26 27.5V32.5M30 27.5V32.5M19 22H37M35 22V33.2C35 34.8802 35 35.7202 34.673 36.362C34.3854 36.9265 33.9265 37.3854 33.362 37.673C32.7202 38 31.8802 38 30.2 38H25.8C24.1198 38 23.2798 38 22.638 37.673C22.0735 37.3854 21.6146 36.9265 21.327 36.362C21 35.7202 21 34.8802 21 33.2V22"
          stroke="#D92D20"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
        <rect
          height="48px"
          rx="24"
          stroke="#FEF3F2"
          stroke-width="8"
          width="48px"
          x="4"
          y="4"
        />
      </svg>
      <p
        class="chakra-text css-0"
      >
        Ooops
      </p>
    </div>
  </header>
  <div
    class="chakra-modal__body css-1scmach"
    id="chakra-modal--body-:r0:"
  >
    Something went wrong
  </div>
  <footer
    class="chakra-modal__footer css-1t7v26l"
  >
    <button
      class="chakra-button css-9rt2sq"
      data-testid="click-confirm"
      type="button"
    >
      Close
    </button>
  </footer>
</section>
`;
